client/packages/react/src/InstantReactAbstractDatabase.ts

```diff
@@ -57,6 +57,195 @@ export type TypingIndicatorHandle<PresenceShape> = {
 
 export const defaultActivityStopTimeout = 1_000;
 
+// ------
+// Topics
+
+function useTopicEffect<
+  RoomSchema extends RoomSchemaShape,
+  RoomType extends keyof RoomSchema,
+  TopicType extends keyof RoomSchema[RoomType]["topics"],
+>(
+  room: InstantReactRoom<any, RoomSchema, RoomType>,
+  topic: TopicType,
+  onEvent: (
+    event: RoomSchema[RoomType]["topics"][TopicType],
+    peer: RoomSchema[RoomType]["presence"],
+  ) => any,
+): void {
+  useEffect(() => {
+    const unsub = room._core._reactor.subscribeTopic(
+      room.id,
+      topic,
+      (event, peer) => {
+        onEvent(event, peer);
+      },
+    );
+
+    return unsub;
+  }, [room.id, topic]);
```

Comment:
I was definitely surprised here: looks like when we subscribe, we don't actually care about the room type. This will probably lead to bugs when a user tries to connect to different room types with the same id. 

Noting to look deeper into this